src/pages/index.js

```diff
@@ -0,0 +1,314 @@
+
+// Load Images
+document.getElementById("pencil").src = pencilSrc;
+document.getElementById("logo").src = logoSrc;
+document.getElementById("plus").src = plusSrc;
+document.getElementById("bessie-coleman").src = colemanSrc;
+
+// Handle Add Card
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+
+  const inputValues = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  setButtonText(evt.submitter, "Saving...");
+
+  api
+    .addCard(inputValues)
+    .then((cardData) => {
+      const cardElement = getCardElement(cardData);
+      cardList.prepend(cardElement);
+      closeModal(cardModal);
+      evt.target.reset();
+      disableButton(cardSubmitButton, settings);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(evt.submitter, false, "Saving...", "Save");
+    });
+}
+
+// Handle Like Card
+function handleLike(evt, id) {
+  const isLiked = evt.target.classList.contains("card__like-btn_liked");
+  evt.target.classList.toggle("card__like-btn_liked");
+  api
+    .changeLikeStatus(id, isLiked)
+    .then(() => {
+      evt.target.classList.toggle("card__like-btn_active");
+    })
+    .catch(console.error);
+}
+
+// Handle Delete Card
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(evt.submitter, true, "Delete", "Deleting..."); // Set loading state
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      selectedCard = null;
+      selectedCardId = null;
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(evt.submitter, false, "Delete")); // Reset button text
+}
+
+// Handle Edit Profile
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  setButtonText(evt.submitter, true, "Save", "Saving..."); // Set loading state
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => setButtonText(evt.submitter, false, "Save")); // Reset button text
+}
+
+// Handle Avatar Update
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(evt.submitter, "Saving...");
+
+  api
+    .editAvatarInfo(avatarLinkInput.value)
+    .then((data) => {
+      console.log(data);
+      if (data.avatar) {
+        document.querySelector(".profile__avatar").src = data.avatar;
+      } else {
+        console.error("Avatar update failed:", data);
+      }
+      closeModal(avatarModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(evt.submitter, false, "Saving...", "Save");
```

Comment:
`"Saving...", "Save"` are not needed here. There is a mistake in the order. It should be `"Save", "Saving..."`


But the 3rd and 4th arguments are stated by **default** inside `setButtonText`, so you don't need to repeat it here


```js
    .finally(() => {
      setButtonText(evt.submitter, false);
    });

```